<a href="https://colab.research.google.com/github/Jannath19/datasets/blob/main/HHA_505_Assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!git clone https://github.com/Jannath19/datasets.git


Cloning into 'datasets'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 14 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (14/14), 3.21 MiB | 14.38 MiB/s, done.


In [2]:
import os

# List files in the "datasets" directory
os.listdir('datasets')


['2017 List of SSA Approved Forms.xlsx',
 'Air_Quality.csv',
 'Housing - Total Housing Estimates.json',
 'supplement22.xlsx',
 'Households - Decennial Census Average Household Sizes.json',
 'SupplyChainEmissionFactorsforUSIndustriesCommodities v1.1.1.xlsx',
 'Community Services Resources - Total Parks and Acreage.json',
 'Mental_Health_Care_in_the_Last_4_Weeks.csv',
 '.git',
 'README.md',
 'diabetes_all_2016.csv']

In [3]:
!ls

!git clone https://github.com/Jannath19/datasets.git


datasets  sample_data
fatal: destination path 'datasets' already exists and is not an empty directory.


In [4]:
%cd datasets


/content/datasets


In [5]:
import pandas as pd

# Define the GitHub raw file URL for your CSV dataset
csv_url = 'https://raw.githubusercontent.com/Jannath19/datasets/main/diabetes_all_2016.csv'

# Load the CSV dataset into a DataFrame
df = pd.read_csv(csv_url)

# Display the first 10 rows
df.head(10)


CT  BPAD  BPAN  BPAN2  BWAD  BWAN  BWAN2  BMAD  BMAN  BMAN2
0  100500  1237   179    130   712    97     72   525    82     58
1  101100  1080    97     65   583    48     33   497    49     32
2  101400  1865   157    118  1033    76     57   832    81     61
3  101600   698    66     44   397    48     35   301    18      9
4  101700   928    87     65   555    54     44   373    33     21
5  101800  1477   145    111   827    81     64   650    64     47
6   10300   674    50     36   299    25     20   375    25     16
7  110200  1930   121     85  1054    53     37   876    68     48
8  110600  1477    79     61   813    43     35   664    36     26
9  111300  1221   115     72   704    67     41   517    48     31

In [6]:
import pandas as pd

# Define the GitHub raw file URL for your CSV dataset
mental_health_csv_url = 'https://raw.githubusercontent.com/Jannath19/datasets/main/Mental_Health_Care_in_the_Last_4_Weeks.csv'

# Load the CSV dataset into a DataFrame
mental_health_df = pd.read_csv(mental_health_csv_url)

# Display the first 10 rows
mental_health_df.head(10)


Indicator              Group  \
0  Took Prescription Medication for Mental Health...  National Estimate   
1  Took Prescription Medication for Mental Health...             By Age   
2  Took Prescription Medication for Mental Health...             By Age   
3  Took Prescription Medication for Mental Health...             By Age   
4  Took Prescription Medication for Mental Health...             By Age   
5  Took Prescription Medication for Mental Health...             By Age   
6  Took Prescription Medication for Mental Health...             By Age   
7  Took Prescription Medication for Mental Health...             By Age   
8  Took Prescription Medication for Mental Health...             By Sex   
9  Took Prescription Medication for Mental Health...             By Sex   

           State            Subgroup Phase  Time Period  \
0  United States       United States     2           13   
1  United States       18 - 29 years     2           13   
2  United States       30 - 39 years     2           13   
3  United States       40 - 49 years     2           13   
4  United States       50 - 59 years     2           13   
5  United States       60 - 69 years     2           13   
6  United States       70 - 79 years     2           13   
7  United States  80 years and above     2           13   
8  United States                Male     2           13   
9  United States              Female     2           13   

       Time Period Label Time Period Start Date Time Period End Date  Value  \
0  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   19.4   
1  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   18.7   
2  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   18.3   
3  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   20.4   
4  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   21.2   
5  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   19.7   
6  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   18.4   
7  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   13.8   
8  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   13.7   
9  Aug 19 - Aug 31, 2020             08/19/2020           08/31/2020   24.6   

   LowCI  HighCI Confidence Interval Quartile Range  Suppression Flag  
0   19.0    19.8         19.0 - 19.8            NaN               NaN  
1   17.2    20.3         17.2 - 20.3            NaN               NaN  
2   17.3    19.2         17.3 - 19.2            NaN               NaN  
3   19.5    21.3         19.5 - 21.3            NaN               NaN  
4   20.2    22.2         20.2 - 22.2            NaN               NaN  
5   18.7    20.7         18.7 - 20.7            NaN               NaN  
6   17.0    19.9         17.0 - 19.9            NaN               NaN  
7   11.0    17.0         11.0 - 17.0            NaN               NaN  
8   13.0    14.3         13.0 - 14.3            NaN               NaN  
9   24.0    25.2         24.0 - 25.2            NaN               NaN

In [7]:
import pandas as pd

# Define the GitHub raw file URL for your CSV dataset
air_quality_csv_url = 'https://raw.githubusercontent.com/Jannath19/datasets/main/Air_Quality.csv'

# Load the CSV dataset into a DataFrame
air_quality_df = pd.read_csv(air_quality_csv_url)

# Display the first 10 rows
air_quality_df.head(10)


Unique ID  Indicator ID                  Name Measure Measure Info  \
0     216498           386            Ozone (O3)    Mean          ppb   
1     216499           386            Ozone (O3)    Mean          ppb   
2     219969           386            Ozone (O3)    Mean          ppb   
3     219970           386            Ozone (O3)    Mean          ppb   
4     164876           383  Sulfur Dioxide (SO2)    Mean          ppb   
5     164877           383  Sulfur Dioxide (SO2)    Mean          ppb   
6     219971           386            Ozone (O3)    Mean          ppb   
7     219972           386            Ozone (O3)    Mean          ppb   
8     164878           383  Sulfur Dioxide (SO2)    Mean          ppb   
9     164879           383  Sulfur Dioxide (SO2)    Mean          ppb   

  Geo Type Name  Geo Join ID                          Geo Place Name  \
0            CD          313                     Coney Island (CD13)   
1            CD          313                     Coney Island (CD13)   
2       Borough            1                                   Bronx   
3       Borough            1                                   Bronx   
4            CD          211        Morris Park and Bronxdale (CD11)   
5            CD          212    Williamsbridge and Baychester (CD12)   
6       Borough            2                                Brooklyn   
7       Borough            2                                Brooklyn   
8            CD          301       Greenpoint and Williamsburg (CD1)   
9            CD          302  Fort Greene and Brooklyn Heights (CD2)   

      Time Period  Start_Date  Data Value  Message  
0     Summer 2013  06/01/2013       34.64      NaN  
1     Summer 2014  06/01/2014       33.22      NaN  
2     Summer 2013  06/01/2013       31.25      NaN  
3     Summer 2014  06/01/2014       31.15      NaN  
4  Winter 2008-09  12/01/2008        5.89      NaN  
5  Winter 2008-09  12/01/2008        5.75      NaN  
6     Summer 2009  06/01/2009       26.27      NaN  
7     Summer 2010  06/01/2010       33.83      NaN  
8  Winter 2008-09  12/01/2008        4.33      NaN  
9  Winter 2008-09  12/01/2008        4.41      NaN

# XLSX **Datasets**

In [8]:
import pandas as pd

# Define the GitHub raw file URL for your XLSX dataset
xlsx_url = 'https://github.com/Jannath19/datasets/raw/main/supplement22.xlsx'

# Load the XLSX dataset into a DataFrame
xlsx_df = pd.read_excel(xlsx_url)

# Display the first 10 rows
xlsx_df.head(10)


2.A OASDI: Coverage, Financing, and Insured Status  \
0  Table 2.A1—Covered employment and self-employm...   
1                                       Year enacted   
2                                                NaN   
3                                               1935   
4                                               1939   
5                                               1946   
6                                               1950   
7                                                NaN   
8                                                NaN   
9                                               1951   

                          Unnamed: 1  \
0                                NaN   
1       Coverage election\nor waiver   
2                                NaN   
3                              . . .   
4                              . . .   
5                              . . .   
6                              . . .   
7               Elective by employer   
8  Elective by employer and employee   
9                              . . .   

                                          Unnamed: 2  
0                                                NaN  
1                                          Provision  
2                                                NaN  
3  All workers in commerce and industry (except r...  
4                        Age restriction eliminated.  
5  Railroad and Social Security earnings combined...  
6  Regularly employed farm and domestic workers. ...  
7  State and local government employees not under...  
8  Employees (other than members of the clergy) o...  
9  Railroad workers with less than 10 years of se...

In [9]:
import pandas as pd

# Define the GitHub raw file URL for your XLSX dataset
xlsx_url = 'https://github.com/Jannath19/datasets/raw/main/SupplyChainEmissionFactorsforUSIndustriesCommodities%20v1.1.1.xlsx'

# Load the XLSX dataset into a DataFrame
xlsx_df = pd.read_excel(xlsx_url)

# Display the first 10 rows
xlsx_df.head(10)


Name  \
0      Authors   
1  Description   
2          DOI   

  Supply Chain GHG Emission Factors for US Commodities and Industries v1.1.1  
0                            Wesley Ingwersen, Mo Li                          
1  Tables presenting supply chain and margin emis...                          
2                    http://doi.org/10.23719/1524524

In [10]:
import pandas as pd

# Define the GitHub raw file URL for your XLSX dataset
xlsx_url = 'https://github.com/Jannath19/datasets/raw/main/2017%20List%20of%20SSA%20Approved%20Forms.xlsx'

# Load the XLSX dataset into a DataFrame
xlsx_df = pd.read_excel(xlsx_url)

# Display the first 10 rows
xlsx_df.head(10)


Form Number                                         Form Title
0     SSA-L8125-F6  Supplemental Security Income Notice Of Interim...
1              NaN                                                NaN
2              NaN                                                NaN
3      SSA-8150-EV      Reporting Events Supplemental Security Income
4              NaN                                                NaN
5              NaN                                                NaN
6   SSA-8150-EV-SP      Reporting Events Supplemental Security Income
7              NaN                                                NaN
8              NaN                                                NaN
9  SSA-L8151-CL-SP                     Sp Cov Lr For SSI Not Of Chang

# JSON **Datasets**

In [11]:
import pandas as pd
import json
import requests

# Define the GitHub raw file URL for your JSON dataset
json_url = 'https://github.com/Jannath19/datasets/raw/main/Housing%20-%20Total%20Housing%20Estimates.json'

# Load the JSON dataset using requests
response = requests.get(json_url)
data = response.json()

# Convert the JSON data to a DataFrame
json_df = pd.DataFrame(data)

# Display the first 10 rows
json_df.head(10)


Profile Year  Housing Unit Estimate  Housing Growth
0          2019                 115400           0.095
1          2020                 117300           0.110
2          2021                 119700           0.136

In [12]:
import pandas as pd
import json
import requests

# Define the GitHub raw file URL for your JSON dataset
json_url = 'https://github.com/Jannath19/datasets/raw/main/Households%20-%20Decennial%20Census%20Average%20Household%20Sizes.json'

# Load the JSON dataset using requests
response = requests.get(json_url)
data = response.json()

# Convert the JSON data to a DataFrame
json_df = pd.DataFrame(data)

# Display the first 10 rows
json_df.head(10)


YEAR  Average
0  1950     3.35
1  1960     2.90
2  1970     2.43
3  1980     2.07
4  1990     2.12
5  2000     2.05
6  2010     2.09
7            NaN

In [13]:
import pandas as pd
import json
import requests

# Define the GitHub raw file URL for your JSON dataset
json_url = 'https://github.com/Jannath19/datasets/raw/main/Community%20Services%20Resources%20-%20Total%20Parks%20and%20Acreage.json'

# Load the JSON dataset using requests
response = requests.get(json_url)
data = response.json()

# Convert the JSON data to a DataFrame
json_df = pd.DataFrame(data)

# Display the first 10 rows
json_df.head(10)


Profile Year  County Owned Parkland (Acreage)  NOVA Parks (Acreage)  \
0        2019.0                            925.0                 136.0   
1        2020.0                            925.0                 136.0   
2        2021.0                            924.0                 136.0   
3           NaN                              NaN                   NaN   

   National Parks Service (Acreage)  County owned parks  NOVA Owned Parks  
0                             417.0               148.0               3.0  
1                             417.0               149.0               3.0  
2                             417.0               148.0               3.0  
3                               NaN                 NaN               NaN

# Markdown

# Diabetes all 2016 data (CSV Data 1)




**Dataset: Diabetes Data**

- **CT (Column 1):**
   - **Description:** This column contains identification or code for each row.

- **BPAD (Columns 2, 3, 4):**
   - **Description:** These columns likely represent blood pressure measurements. Column 2 may represent systolic blood pressure, column 3 may represent diastolic blood pressure, and column 4 may represent another aspect of blood pressure.

- **BWAD (Columns 5, 6, 7):**
   - **Description:** These columns may be related to body weight measurements.

- **BMAD (Columns 8, 9, 10):**
   - **Description:** These columns may be associated with body mass or other measurements, but not exactly clear



# Mental health care in the last 4 weeks data (CSV Data 2)

**Dataset: Mental Health Medication Usage**

- **Indicator:**
   - **Description:** Indicates the specific health-related indicator or measurement being recorded.

- **Group:**
   - **Description:** The group or category to which the data belongs.

- **State:**
   - **Description:** The geographic location or region associated with the data.

- **Subgroup:**
   - **Description:** A subcategory or subset within the group.

- **Phase:**
   - **Description:** Represents a phase or stage related to data collection.

- **Time Period:**
   - **Description:** Indicates the time period during which the data was collected.

- **Time Period Label:**
   - **Description:** A label associated with the time period.

- **Value:**
   - **Description:** The numeric value or measurement for the indicator.
  
- **LowCI (Low Confidence Interval):**
   - **Description:** The lower bound of the confidence interval for the value.
  

- **HighCI (High Confidence Interval):**
   - **Description:** The upper bound of the confidence interval for the value.
   

- **Confidence Interval:**
   - **Description:** A range of values within which the true value is likely to fall.

- **Quartile Range:**
   - **Description:** The range between the first quartile (25th percentile) and the third quartile (75th percentile).

- **Suppression Flag:**
   - **Description:** A flag indicating whether certain data points are suppressed or not reported.



# Air Quality data (CSV data 3)

**Dataset: Air Quality Measurements**

- **Unique ID:**
   - **Description:** A unique identifier for each data point.

- **Indicator ID:**
   - **Description:** An identifier associated with the specific air quality indicator.

- **Name:**
   - **Description:** The name of the air quality indicator.
   - Example: Ozone (O3)

- **Measure:**
   - **Description:** The type of measurement being recorded (e.g., Mean).

- **Measure Info:**
   - **Description:** Additional information about the measurement (e.g., "ppb" for parts per billion).

- **Geo Type Name:**
   - **Description:** The type of geographic region or area being measured (e.g., "CD" for Community District).

- **Geo Join ID:**
   - **Description:** An identifier associated with the geographic area.

- **Geo Place Name:**
   - **Description:** The name of the geographic location or place.
   - Example: Coney Island (CD13)

- **Time Period:**
   - **Description:** The time period during which the measurement was taken.

- **Start Date:**
   - **Description:** The start date of the measurement period.

- **Data Value:**
   - **Description:** The numeric value or measurement of the air quality indicator.

- **Message:**
   - **Description:** An optional message or comment related to the data point



# Supplement Data (XLSX data 1)

**Dataset: Social Security Covered Employment and Self-Employment Provisions**

- **Year Enacted:**
   - **Description:** The year in which various Social Security covered employment and self-employment provisions were enacted.

- **Coverage Election or Waiver:**
   - **Description:** Information related to the coverage election or waiver associated with each provision, as it evolved over time.

- **Provision:**
   - **Description:** Detailed information about the Social Security provisions, including coverage criteria, restrictions, and any changes made over the years.

Dataset Description:
This dataset, Covered Employment and Self-Employment Provisions, by Year Enacted, provides a historical record of Social Security provisions related to covered employment and self-employment. It includes details about when these provisions were enacted, how coverage elections or waivers evolved, and the specific criteria and conditions associated with each provision.




## Supply chain emission factors for US industries commodities (XLSX data 2) (This had a lot of datasets; I primarily focused on 2011 detail commodity)

**Dataset: Supply Chain Emission Factors for US Industries Commodities**

- **Commodity Code:**
   - **Description:** A unique code representing different commodities.
   
- **Commodity Name:**
   - **Description:** The name of the commodity associated with the code.

- **Substance:**
   - **Description:** The type of substance (e.g., carbon dioxide, methane, nitrous oxide) emitted during the supply chain of the commodity.

- **Unit:**
   - **Description:** The unit of measurement used to quantify emission factors (e.g., kg/2018 USD, purchaser price).

- **Supply Chain Emission Factors without Margins:**
   - **Description:** Emission factors representing the quantity of emitted substance per unit of measurement without considering margins.

- **Margins of Supply Chain Emission Factors:**
   - **Description:** Marginal values associated with the emission factors.

- **Supply Chain Emission Factors with Margins:**
   - **Description:** Emission factors considering both the quantity of emitted substance and margins.

- **DQ Reliability Score of Factors without Margins:**
   - **Description:** A reliability score indicating the quality of data related to emission factors without considering margins.

- **DQ Temporal Correlation of Factors without Margins:**
   - **Description:** Information about the temporal correlation of emission factors without considering margins.

- **DQ Geographical Correlation of Factors without Margins:**
   - **Description:** Information about the geographical correlation of emission factors without considering margins.

- **DQ Technological Correlation of Factors without Margins:**
   - **Description:** Information about the technological correlation of emission factors without considering margins.

- **DQ Data Collection of Factors without Margins:**
   - **Description:** Information about data collection related to emission factors without considering margins.



# 2017 list of SSA approved forms data (XLSX data 3)

**Dataset: SSA Approved Forms**

- **Form Number:**
   - **Description:** The unique identification number assigned to each Social Security Administration (SSA) approved form.

- **Form Title:**
   - **Description:** The title or name of the SSA-approved form.


Total housing estimates data (JSON data 1)


**Dataset: Housing Unit Estimates**

- **Profile Year:**
   - **Description:** The year for which the housing unit estimate is provided.

- **Housing Unit Estimate:**
   - **Description:** The estimated number of housing units for the corresponding profile year.

- **Housing Growth:**
   - **Description:** The percentage increase in housing units compared to the previous year.



# Decennial census average household sizes data (JSON data 2)

**Dataset: Average Family Size Over Time**

- **YEAR:**
   - **Description:** The year for which the average family size is recorded.

- **Average:**
   - **Description:** The average family size for the corresponding year.


- **Data:**
In 1950, the average family size was 3.35.
- By 1960, the average family size decreased to 2.9.
- In 1970, the average family size further declined to 2.43.

(The dataset provides insights into changes in average family size over time and can be useful for demographic studies, social research, and trend analysis related to family structures and sizes.)


# Community services resources - total parks and acreage data (JSON data 3)

**Dataset: Parkland Acreage by Ownership (2019-2021)**

- **Profile Year:**
   - **Description:** The year for which parkland acreage data is recorded.

- **County Owned Parkland (Acreage):**
   - **Description:** The total acreage of parkland owned by the county for the corresponding year.

- **NOVA Parks (Acreage):**
   - **Description:** The total acreage of parkland owned by NOVA Parks (assuming NOVA refers to a park authority or organization) for the corresponding year.

- **National Parks Service (Acreage):**
   - **Description:** The total acreage of parkland managed by the National Parks Service for the corresponding year.

- **County Owned Parks:**
   - **Description:** The number of county-owned parks for the corresponding year.

- **NOVA Owned Parks:**
   - **Description:** The number of parks owned by NOVA Parks (assuming NOVA refers to a park authority or organization) for the corresponding year.


(The dataset contains an additional record with null values for all columns, possibly indicating missing or incomplete data for a specific year.)